c7n/resources/sns.py

```diff
@@ -47,6 +48,100 @@ class resource_type(object):
             'SubscriptionsDeleted'
         )
 
+    permissions = ('SNS:ListTagsForResource',)
+
+    def augment(self, resources):
+        client = local_session(self.session_factory).client('sns')
+
+        def _augment(r):
+            tags = self.retry(client.list_tags_for_resource,
+                ResourceArn=r['TopicArn'])['Tags']
+            r['Tags'] = tags
+            return r
+
+        resources = super(SNS, self).augment(resources)
+        with self.executor_factory(max_workers=3) as w:
+            return list(w.map(_augment, resources))
+
+
+@SNS.action_registry.register('tag')
+class TagSNS(Tag):
+    """Action to create tag(s) on sns
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: tag-sns
+                resource: sns
+                filters:
+                  - "tag:target-tag": absent
+                actions:
+                  - type: tag
+                    key: target-tag
+                    value: target-tag-value
+    """
+
+    permissions = ('sns:TagResource',)
+
+    def process_resource_set(self, client, resources, new_tags):
+        for r in resources:
+            try:
+                client.tag_resource(
+                    ResourceArn=r['TopicArn'],
+                    Tags=new_tags)
+            except client.exceptions.ResourceNotFound:
+                continue
+
+
+@SNS.action_registry.register('remove-tag')
+class UntagMessageBroker(RemoveTag):
```

Comment:
"MessageBroker" looks like a copy/paste issue... Topic is more appropriate for sns.

Reply:
My fault, changed all three class names to Topic